### ISSUE TITLE
CompileBPFFilter functionality in pure go

### ISSUE THREAD

**Issue by mosajjal** [google/gopacket#851]
Hi there,

I was wondering if there's a pure go implementation of `pcap`'s CompileBPFFilter functionality. This one function is the only barrier that I have to convert my project to pure go. I took a look at a bunch of projects and it looks like the handler of `tcpdump`-like BPF filter is always imported from `libpcap`. 

is this something `gopacket` will consider building and maintaing as part of the project or you see it as something an external go library can potentially handle.

Cheers,

**Comment by gconnell** [comment]
I'd be more than happy to accept a pull request, but I don't have the expertise to build myself.

**Comment by yerden** [comment]
There is a pure-Go implementation of BPF virtual machine [here](https://pkg.go.dev/golang.org/x/net/bpf). I played with it at some point and it was functional. Eventually I gave up the idea of using it instead of libpcap's.

The first reason was that performance was low compared to libpcap's version which is in C. Moreover, `pcap_setfilter` may take advantage of some custom BPF execution environment, e.g. hardware acceleration of some sort, which will make the performance gap even more palpable.

The second reason was that this would add extra complexity to my project without cutting the dependence on libpcap.
